test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java

```diff
@@ -457,6 +462,72 @@ public void checkpointWriterInterleavedTest() throws Exception {
         }
     }
 
+    /** Test the CheckpointWriter write size limit.
+     *
+     * CheckpointWriter aggregates a batch of 50 SMREntries into one
+     * CheckpointEntry. This test uses large-sized SMREntries to verify
+     * that batching will not violate the maxWriteSize limit.
+     */
+    @Test
+    public void checkpointWriterSizeLimitTest() throws Exception {
+        final String streamName = "mystream5";
+        final UUID streamId = CorfuRuntime.getStreamID(streamName);
+        final String keyPrefix = "a-prefix";
+        final int numKeys = 50;
+        final String author = "Me, myself, and I";
+        final String mutationSuffix = "_mutation_suffix";
+        final int base = 1024;
+        // max write size set to 25MB
+        final int maxWriteSize = 26214400;
+
+        StreamingMap<String, String> m = instantiateStringMap(streamName);
+
+        // each entry is 1MB
+        byte[] buf = new byte[base * base];
+        ThreadLocalRandom.current().nextBytes(buf);
+        String payload = new String(buf);
+
+        for (int i = 0; i < numKeys; i++) {
+            m.put(keyPrefix + i, payload);
+        }
+
+        // Set up CP writer.  Add fudgeFactor to all CP data,
```

Comment:
what does this comment mean about the fudgefactor?

Reply:
@Lujie1996 can better answer this one. George?

Reply:
Removed.